compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1461,6 +1461,14 @@ impl<'tcx> Place<'tcx> {
         self.projection.iter().any(|elem| elem.is_indirect())
     }
 
+    /// If MirPhase >= Derefered and if projection contains Deref,
+    /// It's guaranteed to be in the first place
+    pub fn has_deref(&self) -> bool {
```

Comment:
This is super confusing, we now have `is_indirect` and `has_deref` and they are basically checking the same thing except one assumes we are in a certain MIR phase and the other doesn't. These doc comments of both methods should explain when to use one vs the other.

Reply:
I wrote this function, I don't quite remember the details why we needed to check if `Deref` was in first place or not but I remember it making quite big difference.